packages/shared_preferences/shared_preferences/test/devtools_extension_test.dart

```diff
@@ -0,0 +1,324 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:shared_preferences/src/devtools_extension.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_async_platform_interface.dart';
+
+import 'shared_preferences_async_test.dart';
+
+typedef _Event = (String eventKind, Map<String, Object?> eventData);
+
+class _FakePostEvent {
+  final List<_Event> eventLog = <_Event>[];
+
+  void call(
+    String eventKind,
+    Map<String, Object?> eventData,
+  ) {
+    eventLog.add((eventKind, eventData));
+  }
+}
+
+void main() {
+  group('DevtoolsExtension', () {
+    late SharedPreferencesAsync asyncPreferences;
+    late _FakePostEvent fakePostEvent;
+    late DevtoolsExtension extension;
+
+    setUp(() {
+      SharedPreferencesAsyncPlatform.instance = FakeSharedPreferencesAsync();
+      asyncPreferences = SharedPreferencesAsync();
+      fakePostEvent = _FakePostEvent();
+      extension = DevtoolsExtension(fakePostEvent.call);
+    });
+
+    test('should request all keys', () async {
+      SharedPreferences.setMockInitialValues(<String, Object>{
+        'key1': 1,
+        'key2': true,
+      });
+      await asyncPreferences.setBool('key3', true);
+      await asyncPreferences.setInt('key4', 1);
+
+      await extension.requestAllKeys();
+
+      expect(fakePostEvent.eventLog.length, equals(1));
+      final (
+        String eventKind,
+        Map<String, Object?> eventData,
+      ) = fakePostEvent.eventLog.first;
+      expect(
+        eventKind,
+        equals('shared_preferences:all_keys'),
+      );
+      expect(
+        eventData,
+        equals(<String, List<String>>{
+          'asyncKeys': <String>['key3', 'key4'],
+          'legacyKeys': <String>['key1', 'key2'],
+        }),
+      );
+    });
+
+    test('should request value from async api', () async {
```

Comment:
Consider putting all legacy API tests in a testing group and all async API tests in a testing group